src/main/java/org/opensrp/web/rest/ManifestResource.java

```diff
@@ -144,4 +152,26 @@ public void setManifestService(ManifestService manifestService) {
                 HttpStatus.OK);
     }
 
+    protected Manifest generateManifest(String jsonString) {
+        Manifest latestManifest = manifestService.getAllManifest(1).get(0);
+        String identifier = "0.0.1";
+        Manifest generatedManifest = new Manifest();
+        if (latestManifest != null) {
+            String json = latestManifest.getJson();
+            if (StringUtils.isNotBlank(json)) {
+                JSONObject jsonObject = new JSONObject(json);
+                if (jsonObject.has(IDENTIFIER)) {
+                    String version = jsonObject.getString(IDENTIFIER);
+                    if (StringUtils.isNotBlank(version)) {
+                       identifier = FormConfigUtils.getNewVersion(version);
+                    }
+                }
+            }
```

Comment:
I don't think you need to generate the form version from the `json identifier`.  Just check if the latest manifest has an identifier if yes then just pass that to the version generator.